### PR TITLE
PanelGridItemRepeater: Refactor 

### PR DESCRIPTION
**Pull request by dprokop**
This PR refactors the usage of `SceneGridItem` and `PanelRepeaterGridItem` into a single scene object responsible for handling repeated and non-repeated panels: `DashboardGridItem`. It's aimed at simplifying the logic for manipulating repeat options, that no longer need to to create/re-create different scenes object structures to handle repeat options change. 

It does not bring too much logic changes, this PR turned out to be mostly about conditions changes, and renames. But I hope to soon do a next step which tries to unify the difference betwiin `VizPanel` and `LibraryVizPanel` to avoid constant instance checks we have all over the place, that bring even more complexity to pretty much every operation agains a dashboard panel.

TODO:
- [x] Fix a bug with saving library panel - the generated model i incorrect
- [x] Bug: Repeated library panels share the same parent and warning in console is shown

### CODE REVIEW

File: public/app/features/library-panels/state/api.ts [hunk 1]
```diff
@@ -138,19 +139,38 @@ export async function getConnectedDashboards(uid: string): Promise<DashboardSear
...
+    ),
     kind: LibraryElementKind.Panel,
   };
   return saveModel;
 }
 
 export async function updateLibraryVizPanel(libraryPanel: LibraryVizPanel): Promise<LibraryElementDTO> {
   const { uid, folderUID, name, model, version, kind } = libraryVizPanelToSaveModel(libraryPanel);
+
+  console.log('updateLibraryVizPanel', model);
```
Comment (torkelo) [inline public/app/features/library-panels/state/api.ts:173]: console.log that remain

File: public/app/features/dashboard-scene/panel-edit/PanelEditor.tsx [hunk 2]
```diff
@@ -105,81 +105,17 @@ export class PanelEditor extends SceneObjectBase<PanelEditorState> {
...
-      source: panelManager.getPanelCloneWithData(),
-      variableName: panelManager.state.repeat!,
-      repeatedPanels: [],
-      repeatDirection: panelManager.state.repeatDirection,
-      maxPerRow: panelManager.state.maxPerRow,
-    });
-    gridLayout.setState({
-      children: gridLayout.state.children.map((child) => (child.state.key === gridItem.state.key ? repeater : child)),
-    });
-  }
```
Comment (torkelo) [inline public/app/features/dashboard-scene/panel-edit/PanelEditor.tsx:156]: nice! ❤️

**Review by torkelo — APPROVED** [review]
This is looking really good! ❤️ 

I did not take it for a test run, still pretty sick and taking it easy

File: public/app/features/dashboard-scene/panel-edit/PanelEditor.tsx [hunk 3]
```diff
@@ -105,81 +105,17 @@ export class PanelEditor extends SceneObjectBase<PanelEditorState> {
...
-      if (repeatToNormal) {
-        this.replacePanelRepeaterWithGridItem(sourcePanelParent);
-      } else {
-        this.handleRepeatOptionChanges(sourcePanelParent);
-      }
+    const gridItem = sourcePanelParent instanceof LibraryVizPanel ? sourcePanelParent.parent : sourcePanelParent;
+
+    if (gridItem instanceof DashboardGridItem) {
+      this.handleRepeatOptionChanges(gridItem);
+      // }
```
Comment (oscarkilhed) [inline public/app/features/dashboard-scene/panel-edit/PanelEditor.tsx:112]:
```suggestion
```

**Review by oscarkilhed — APPROVED** [review]
Looks good.

Have tested editing panels, repeating panels, library panels and repeating library panels. Seems to work!

**Review by ivanortegaalba — APPROVED** [review]
This is very good work removing duplicity and making the treatment of the object easier. This is good to go! 💯 

There is a test failing, probably need some tweaks

**Review by mdvictor — APPROVED** [review]
Looks and works great! Awesome work simplifying the repeater. 

I found something but I don't think it's really related. I can add it to the bug list. If you add a repeated panel and then a row to hold it, and then repeat that row with the same variable used for the panel it crashes in the console, but UI _seems_ to work fine

**Pull request merged**